timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py

```diff
@@ -304,6 +318,7 @@ def _fit(
         model_params = self.get_hyperparameters()
 
         mlforecast_init_args = self._get_mlforecast_init_args(train_data, model_params)
+        assert self.freq is not None
```

Comment:
We have this assertion multiple times. Is this intended, or should we maybe move it to `__init__`?

Reply:
Yeah pyright doesn't recognize if the assert is in `__init__`.